src/npcmove.cpp

```diff
@@ -1380,7 +1380,7 @@ npc_action npc::method_of_attack()
 
     // if the best mode is within the confident range try for a shot
     if( g_mode && sees( *critter ) && has_los &&
-        confident_gun_mode_range( g_mode, cur_recoil ) >= dist ) {
+        confident_gun_mode_range( g_mode, cur_recoil ) >= dist && g_mode->gun_range( true ) >= dist ) {
```

Comment:
```suggestion
        g_mode->gun_range( true ) >= dist && confident_gun_mode_range( g_mode, cur_recoil ) >= dist ) {
```

iirc `confident_gun_mode_range` is much more expensive to calculate than getting gun range so short-circuiting calculation could be better.